ExampleMod/Common/GlobalBuffs/ExampleGlobalBuff.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Common.GlobalBuffs
+{
+	//Showcases how to work with all buffs.
+	public class ExampleGlobalBuff : GlobalBuff
+	{
+		public override void Update(int type, Player player, ref int buffIndex) {
+			//If the player gets the Chilled debuff while he already has more than 5 other buffs/debuffs, limit the max duration to 3 seconds.
+			if (type == BuffID.Chilled && buffIndex >= 5) {
+				int limit = 3 * 60;
+				if (player.buffTime[buffIndex] > limit) {
+					player.buffTime[buffIndex] = limit;
+				}
+			}
+		}
+
+		public override bool PreDraw(SpriteBatch spriteBatch, int type, int buffIndex, ref BuffDrawParams drawParams) {
+			//Make the campfire buff have a different color and shake slightly.
+			if (type == BuffID.Campfire) {
+				drawParams.drawColor = Main.DiscoColor * Main.buffAlpha[buffIndex];
+				Vector2 shake = new Vector2(Main.rand.Next(-2, 3), Main.rand.Next(-2, 3));
+				drawParams.position += shake;
+				drawParams.textPosition += shake;
+			}
+
+			//If the buff is one of the three specified:
+			if (System.Array.IndexOf(new int[] { BuffID.Regeneration, BuffID.Ironskin, BuffID.Swiftness }, type) > -1) {
```

Comment:
Constantly creating new arrays like that is not a good practice.
```suggestion
			if (type == BuffID.Regeneration || type == BuffID.Ironskin || type == BuffID.Swiftness) {
```